lib/cosmos/tools/table_manager/table_manager.rb

```diff
@@ -315,640 +385,398 @@ def initialize_menus(no_tables = false)
+    rescue => err
+      ExceptionDialog.new(self, err, "Table Save Errors", false)
     end
 
-    def context_menu(point)
-      begin
-        table = @core.table_def.get_table(@currently_displayed_table_name)
-        gui_table = @gui_tables[@currently_displayed_table_name]
-        if table and gui_table
-          table_item = gui_table.itemAt(point)
-          if table_item
-            menu = Qt::Menu.new()
-
-            if table.type == :TWO_DIMENSIONAL
-              item_name = gui_table.horizontalHeaderItem(table_item.column).text + table_item.row.to_s
-            else
-              item_name = gui_table.verticalHeaderItem(table_item.row).text
-            end
-            details_action = Qt::Action.new(tr("Details"), self)
-            details_action.statusTip = tr("Popup details about #{@currently_displayed_table_name} #{item_name}")
-            details_action.connect(SIGNAL('triggered()')) do
-              TlmDetailsDialog.new(nil,
-                                   'TABLE',
-                                   @currently_displayed_table_name.upcase,
-                                   item_name,
-                                   table)
-            end
-            menu.addAction(details_action)
-
-            default_action = Qt::Action.new(tr("Default"), self)
-            default_action.statusTip = tr("Set item to default value")
-            default_action.connect(SIGNAL('triggered()')) do
-              item = table.get_item(item_name)
-              table.write(item.name, item.default)
-              update_gui_item(@currently_displayed_table_name, table, item, table_item.row, table_item.column)
-            end
-            menu.addAction(default_action)
+    # Menu option to save the currently displayed table to an existing table binary file
+    # containing that table.
+    def table_commit
+      return if abort_on_no_current_table()
+      return if abort_on_modified()
+      save_gui_data(current_table_name)
 
-            global_point = gui_table.mapToGlobal(point)
-            global_point.x += gui_table.verticalHeader.width
-            menu.exec(global_point)
-            menu.dispose
-          end
-        end
-      rescue => err
-        ExceptionDialog.new(self, err, "context_menu Errors", false)
+      bin_path = Qt::FileDialog.getOpenFileName(self, "Open Binary", @bin_path,
+                                                "Binary File (*.bin *.dat);;All Files (*)")
+      return unless bin_path
+      def_path = def_path_from_bin_path(bin_path)
+      def_path = get_best_def_path(def_path, bin_path)
+      if !def_path || !File.file?(def_path)
+        def_path = Qt::FileDialog.getOpenFileName(self, "Open Definition File", def_path,
+                                                  "Definition File (*.txt);;All Files (*)")
       end
+      return unless def_path
+
+      @core.table_commit(current_table_name, bin_path, def_path)
```

Comment:
This no longer automatically opens the new file which you've committed to. I'm not sure how I feel about that. I probably need a dialog or something to say success.